packages/block-editor/src/components/text-align-control/index.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { alignLeft, alignCenter, alignRight } from '@wordpress/icons';
+
+const TEXT_ALIGNMENT_CONTROLS = [
+	{
+		icon: alignLeft,
+		label: __( 'Align text left' ),
+		align: 'left',
+	},
+	{
+		icon: alignCenter,
+		title: __( 'Align text center' ),
+		align: 'center',
+	},
+	{
+		icon: alignRight,
+		title: __( 'Align text right' ),
+		align: 'right',
+	},
+];
+
+const DEFAULT_CONTROLS = [ 'left', 'center', 'right' ];
+
+/**
+ * Control to facilitate text alignment selections.
+ *
+ * @param {Object}   props           Component props.
+ * @param {string}   props.className Class name to add to the control.
+ * @param {string}   props.value     Currently selected text align.
+ * @param {Function} props.onChange  Handles change in text alignment selection.
+ * @param {Array}    props.controls  Array of text align controls to display.
+ *
+ * @return {Element} Text align control.
+ */
+export default function TextAlignControl( {
+	className,
+	value,
+	onChange,
+	controls = DEFAULT_CONTROLS,
+} ) {
+	const validControls = TEXT_ALIGNMENT_CONTROLS.filter( ( control ) =>
+		controls.includes( control.align )
+	);
+
+	if ( ! validControls.length ) {
+		return null;
+	}
```

Comment:
Do you think we need this filtering here given the other text decoration and transform controls didn't?

I believe the display of the others is controlled by theme.json enabling or disabling it. Either way, having them consistent would be good.

On that note, it feels like the three controls could be refactored to just be wrappers for a single control and share some reduced CSS.

Reply:
> Do you think we need this filtering here given the other text decoration and transform controls didn't?
> 
> I believe the display of the others is controlled by theme.json enabling or disabling it. Either way, having them consistent would be good.

My understanding is that `TextDecorationControl` and `TextTransformControl` components cannot override controls, so they are guaranteed not to have errors. On the other hand, since this component can overwrite the control, I think an error will occur when the component is used, for example, as shown below.

```jsx
<TextAlignmentControl
  controls={[ 'left', 'center', 'invalid' ]}
  onChange={() => {}}
/>
```

The available values are clearly stated in the documentation, but I personally think it's okay to have this filter.